JEP.md

```diff
@@ -178,6 +178,46 @@ HTTP PUT /api/telemetry/event
 
 1. Is this work done on the standalone jupyter-server implementation or on the classic jupyter/notebook?
 
+### JupyterHub
+
+JupyterHub would use the same underlying python library as the notebook server, and be configured
+in the same way.
+
+JupyterHub relies on extension through Spawners, Authenticators, Proxies & Services to
+do most of its work. They would have interesting events to emit, so we should make sure
+they have easy ways to.
+
+#### Spawners, Authenticators & Proxies
+
+Spawners, Authenticators & Proxies run in-process with JupyterHub, and have access to
+the JupyterHub `app` object. When they want to emit events, they will:
+
+1. Write schemas for events they want to emit
+2. Register their schemas with the with the eventlogging library
+
+   ```python
+   self.app.event_log.register_schema
+   for schema_file in glob(os.path.join(here, 'event-schemas','*.json')):
+      with open(schema_file) as f:
+          self.event_log.register_schema(json.load(f))
+   ```
+3. Emit events wherever they need.
+
+   ```python
+   self.app.event_log.emit('kubespawner.hub.jupyter.org/pod-missing', 1, {
```

Comment:
Should this not be consistent with the `record_event` API that we have above? Or just JupyterHub have its own concept of the `event_log`

Reply:
Yeah, I marked in the TODO we need to reconcile these two, since they should be the same code. Right now I copy pasted from the JupyterHub PR.

Reply:
SGTM